app/lib/language_mapper.js

```diff
@@ -16,96 +16,126 @@
    along with Ezra Bible App. See the file LICENSE.
    If not, see <http://www.gnu.org/licenses/>. */
 
-class LanguageMapper {
-  constructor() {
-    this.mappingExistsCache = {};
-    this.langs = null;
-  }
 
-  getLangs() {
-    if (this.langs == null) {
-      this.langs = require('iso-639-3');
-    }
+const hasIntlDisplayNames = Intl && typeof Intl === "object" && typeof Intl.DisplayNames === "function";
 
-    return this.langs;
+var mappingExistsCache = {};
+module.exports.getLanguageName = (languageCode, localeCode = 'en') => {
+  if (mappingExistsCache[languageCode] && mappingExistsCache[languageCode][localeCode]) {
+    return mappingExistsCache[languageCode][localeCode];
   }
 
-  mappingMatchesCode(mapping, languageCode) {
-    return (languageCode == mapping.iso6393 ||
-            languageCode == mapping.iso6392B ||
-            languageCode == mapping.iso6392T ||
-            languageCode == mapping.iso6391);
-  }
+  const details = this.getLanguageDetails(languageCode, localeCode);
+  const languageName = details.localized || localeCode === 'en' ? details.languageName : undefined;
 
-  normalizeLanguageCode(languageCode) {
-    var normalizedCode = languageCode.split('-');
-    return normalizedCode[0];
+  if (!mappingExistsCache[languageCode]) {
+    mappingExistsCache[languageCode] = {};
   }
+  mappingExistsCache[languageCode][localeCode] = languageName;
 
-  mappingExists(languageCode) {
-    if (languageCode in this.mappingExistsCache) {
+  return languageName;
+};
 
-      return this.mappingExistsCache[languageCode];
+module.exports.getLanguageDetails = function (languageCode, localeCode = 'en') {
```

Comment:
I suggest to add JSDoc comments - from the outset it's not super clear what this function returns.